offchain/authority-claimer/src/listener.rs

```diff
@@ -70,53 +78,154 @@ impl BrokerListener for DefaultBrokerListener {
     }
 }
 
+// ------------------------------------------------------------------------------------------------
+// MultidappBrokerListener
+// ------------------------------------------------------------------------------------------------
+
+/// The `MultidappBrokerListener` listens for claims from multiple dapps.
+/// It updates its internal list of dapps by consuming from redis' DappsStream.
+#[derive(Debug)]
+pub struct MultidappBrokerListener {
+    broker: Broker,
+    streams: HashMap<RollupsClaimsStream, String>, // stream => last-claim-id
+    chain_id: u64,
+}
+
+impl MultidappBrokerListener {
+    pub async fn new(
+        broker_config: BrokerConfig,
+        chain_id: u64,
+    ) -> Result<Self, BrokerError> {
+        tracing::trace!(
+            "Connecting to the broker ({:?}) on multidapp mode",
+            broker_config
+        );
+        let broker = Broker::new(broker_config).await?;
+        let streams = HashMap::new();
+        Ok(Self {
+            broker,
+            streams,
+            chain_id,
+        })
+    }
+}
+
+impl MultidappBrokerListener {
+    /// Reads addresses from the DappStream and
+    /// converts them to the stream to last-consumed-id map.
+    async fn update_streams(&mut self) -> Result<(), BrokerListenerError> {
+        let initial_id = INITIAL_ID.to_string();
+
+        let streams: Vec<_> =
+            self.broker.get_dapps().await.context(BrokerSnafu)?;
+        tracing::info!("Got the following dapps: {:?}", self.streams);
```

Comment:
Since the `experimental-dapp-addresses-config` key is hardcoded, maybe we could log its name here.

I'm running this code and although I have the stream filled, I can't see this log being updated

```
127.0.0.1:6379> SMEMBERS experimental-dapp-addresses-config
1) "7C54E3f7A8070a54223469965A871fB8f6f88c22"
```

```
 INFO authority_claimer: Creating the multidapp broker listener
 INFO http_server: Starting HTTP server at 0.0.0.0:8080
 INFO authority_claimer::listener: Got the following dapps: {}
```

Reply:
Done. Ping me about the empty list.